## Manatee Mortality Stats from Florida Fish and Wildlife
### Florida Fish & Wildlife Conservation Commission Marine Mammal Pathobiology Laboratory
### 2018 Preliminary Manatee Mortality Table with 5-Year Summary From: 02/01/2018 To: 12/31/2018
### Data obtained from [Florida Fish & Wildlife Conservation Commission](https://myfwc.com/research/manatee/rescue-mortality-response/statistics/mortality/)

In [23]:
import datetime as dt
import pandas as pd

### Convert PDF to spreadsheet using [this website](https://altoconvertpdftoexcel.com/)

In [3]:
df = pd.read_excel('manatees_2018.xlsx', header=None)
df.head()

0   1   2   3   4    5   \
0  FLORIDA FISH AND WILDLIFE CONSERVATION COMMISS... NaN NaN NaN NaN  NaN   
1                                             County NaN NaN NaN NaN  NaN   
2                                             Citrus NaN NaN NaN NaN  NaN   
3                                               Levy NaN NaN NaN NaN  NaN   
4                                            Broward NaN NaN NaN NaN  NaN   

                    6    7         8    9   ...   16   17              18  \
0                  NaN  NaN       NaN  NaN  ...  NaN  NaN             NaN   
1                 Date  NaN  Field ID  NaN  ...  NaN  NaN            City   
2  2018-02-01 00:00:00  NaN  MNW18021  NaN  ...  NaN  NaN   Crystal River   
3  2018-02-01 00:00:00  NaN  MNW18026  NaN  ...  NaN  NaN       Cedar Key   
4  2018-02-01 00:00:00  NaN  MSE18024  NaN  ...  NaN  NaN  Ft. Lauderdale   

    19   20                                   21   22   23  24  25  
0  NaN  NaN                                  NaN  NaN  NaN NaN NaN  
1  NaN  NaN                       Probable Cause  NaN  NaN NaN NaN  
2  NaN  NaN                       Natural: Other  NaN  NaN NaN NaN  
3  NaN  NaN                 Natural: Cold Stress  NaN  NaN NaN NaN  
4  NaN  NaN  Human Related: Watercraft Collision  NaN  NaN NaN NaN  

[5 rows x 26 columns]

In [4]:
df.tail(20)

0   1   2   3   4   \
735                                             Walton NaN NaN NaN NaN   
736                                          St. Lucie NaN NaN NaN NaN   
737                                                Lee NaN NaN NaN NaN   
738                                             Monroe NaN NaN NaN NaN   
739                                            Collier NaN NaN NaN NaN   
740                                            Collier NaN NaN NaN NaN   
741                                                NaN NaN NaN NaN NaN   
742                                                NaN NaN NaN NaN NaN   
743                                                NaN NaN NaN NaN NaN   
744                             For The Period of Time NaN NaN NaN NaN   
745                                 Total=         824 NaN NaN NaN NaN   
746  Mortality comparisons for the time period from... NaN NaN NaN NaN   
747                                               Year NaN NaN NaN NaN   
748                                               2018 NaN NaN NaN NaN   
749                                               2017 NaN NaN NaN NaN   
750                                               2016 NaN NaN NaN NaN   
751                                               2015 NaN NaN NaN NaN   
752                                               2014 NaN NaN NaN NaN   
753                                               2013 NaN NaN NaN NaN   
754                                     5-year average NaN NaN NaN NaN   

             5                    6                7         8   \
735         NaN  2018-12-30 00:00:00              NaN  MNW18166   
736         NaN  2018-12-30 00:00:00              NaN  MSE18098   
737         NaN  2018-12-30 00:00:00              NaN  MSW18340   
738         NaN  2018-12-30 00:00:00              NaN  MSW18341   
739         NaN  2018-12-30 00:00:00              NaN  MSW18343   
740         NaN  2018-12-31 00:00:00              NaN  MSW18342   
741         NaN                  NaN              NaN       NaN   
742         NaN                  NaN              NaN       NaN   
743         NaN                  NaN              NaN       NaN   
744         NaN                  NaN              NaN       NaN   
745         NaN                  NaN              NaN       NaN   
746         NaN                  NaN              NaN       NaN   
747  Watercraft                  NaN  Flood Gate/Lock       NaN   
748   123 (15%)                  NaN                6       NaN   
749   111 (21%)                  NaN                3       NaN   
750   106 (20%)                  NaN                2       NaN   
751    86 (21%)                  NaN                5       NaN   
752    69 (19%)                  NaN                3       NaN   
753     73 (9%)                  NaN                5       NaN   
754    89 (18%)                  NaN                3       NaN   

                           9   ...           16       17          18  \
735                       NaN  ...          NaN      NaN   Niceville   
736                       NaN  ...          NaN      NaN  Ft. Pierce   
737                       NaN  ...          NaN      NaN  Cape Coral   
738                       NaN  ...          NaN      NaN    Flamingo   
739                       NaN  ...          NaN      NaN     Ochopee   
740                       NaN  ...          NaN      NaN      Naples   
741                       NaN  ...          NaN      NaN         NaN   
742                       NaN  ...          NaN      NaN         NaN   
743                       NaN  ...          NaN      NaN         NaN   
744  01/01/2018 -\n12/31/2018  ...          NaN      NaN         NaN   
745                       NaN  ...          NaN      NaN         NaN   
746                       NaN  ...          NaN      NaN         NaN   
747                       NaN  ...  Cold Stress  Natural         NaN   
748                       NaN  ...           79      228         NaN   
749      

In [5]:
# test slice

df.iloc[749:753, 0:]

0   1   2   3   4          5    6  7    8    9   ...  16  17   18  19  \
749  2017 NaN NaN NaN NaN  111 (21%)  NaN  3  NaN  NaN  ...  30  93  NaN  26   
750  2016 NaN NaN NaN NaN  106 (20%)  NaN  2  NaN  NaN  ...  24  86  NaN  36   
751  2015 NaN NaN NaN NaN   86 (21%)  NaN  5  NaN  NaN  ...  18  44  NaN  31   
752  2014 NaN NaN NaN NaN   69 (19%)  NaN  3  NaN  NaN  ...  26  29  NaN  17   

      20   21  22   23  24  25  
749  114  NaN  29  538 NaN NaN  
750  113  NaN  34  520 NaN NaN  
751   95  NaN  24  405 NaN NaN  
752   80  NaN  38  371 NaN NaN  

[4 rows x 26 columns]

In [6]:
clean_df = df.iloc[:741, 0:]
clean_df.tail(10)

0   1   2   3   4    5                    6    7         8   \
731  Hillsborough NaN NaN NaN NaN  NaN  2018-12-27 00:00:00  NaN  MNW18164   
732  Indian River NaN NaN NaN NaN  NaN  2018-12-28 00:00:00  NaN  MEC18150   
733  Hillsborough NaN NaN NaN NaN  NaN  2018-12-29 00:00:00  NaN  MNW18165   
734       Broward NaN NaN NaN NaN  NaN  2018-12-29 00:00:00  NaN  MSE18097   
735        Walton NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MNW18166   
736     St. Lucie NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSE18098   
737           Lee NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSW18340   
738        Monroe NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSW18341   
739       Collier NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSW18343   
740       Collier NaN NaN NaN NaN  NaN  2018-12-31 00:00:00  NaN  MSW18342   

      9   ...   16   17            18   19   20  \
731  NaN  ...  NaN  NaN  Apollo Beach  NaN  NaN   
732  NaN  ...  NaN  NaN     Sebastian  NaN  NaN   
733  NaN  ...  NaN  NaN         Tampa  NaN  NaN   
734  NaN  ...  NaN  NaN     Hollywood  NaN  NaN   
735  NaN  ...  NaN  NaN     Niceville  NaN  NaN   
736  NaN  ...  NaN  NaN    Ft. Pierce  NaN  NaN   
737  NaN  ...  NaN  NaN    Cape Coral  NaN  NaN   
738  NaN  ...  NaN  NaN      Flamingo  NaN  NaN   
739  NaN  ...  NaN  NaN       Ochopee  NaN  NaN   
740  NaN  ...  NaN  NaN        Naples  NaN  NaN   

                                      21   22   23  24  25  
731                 Human Related: Other  NaN  NaN NaN NaN  
732         Undetermined: Too Decomposed  NaN  NaN NaN NaN  
733                       Natural: Other  NaN  NaN NaN NaN  
734         Undetermined: Too Decomposed  NaN  NaN NaN NaN  
735                 Natural: Cold Stress  NaN  NaN NaN NaN  
736         Undetermined: Too Decomposed  NaN  NaN NaN NaN  
737                Perinatal (<= 150 cm)  NaN  NaN NaN NaN  
738  Human Related: Watercraft Collision  NaN  NaN NaN NaN  
739                 Natural: Cold Stress  NaN  NaN NaN NaN  
740         Undetermined: Too Decomposed  NaN  NaN NaN NaN  

[10 rows x 26 columns]

#### Drop NA columns

In [7]:
clean_df.isnull().sum()

0       0
1     741
2     741
3     741
4     741
5     741
6       1
7     741
8       1
9     741
10    741
11    741
12      1
13      3
14    741
15      1
16    741
17    741
18      1
19    741
20    741
21      1
22    741
23    741
24    741
25    741
dtype: int64

In [8]:
# drop columns with a threshold of more than 3 NA values because I'm not sure what the column 13 is about

clean_df.dropna(axis=1, thresh=3, inplace=True)
clean_df.head()

0                    6   \
0  FLORIDA FISH AND WILDLIFE CONSERVATION COMMISS...                  NaN   
1                                             County                 Date   
2                                             Citrus  2018-02-01 00:00:00   
3                                               Levy  2018-02-01 00:00:00   
4                                            Broward  2018-02-01 00:00:00   

         8    12         13                 15              18  \
0       NaN  NaN        NaN                NaN             NaN   
1  Field ID  Sex  Size (cm)           Waterway            City   
2  MNW18021    M        269          Kings Bay   Crystal River   
3  MNW18026    M        202  Black Point Swamp       Cedar Key   
4  MSE18024    U        275    Port Everglades  Ft. Lauderdale   

                                    21  
0                                  NaN  
1                       Probable Cause  
2                       Natural: Other  
3                 Natural: Cold Stress  
4  Human Related: Watercraft Collision

In [9]:
# drop first row and convert second row to headers

clean_df.drop(df.index[0], inplace=True)
clean_df.head()

0                    6         8    12         13                 15  \
1   County                 Date  Field ID  Sex  Size (cm)           Waterway   
2   Citrus  2018-02-01 00:00:00  MNW18021    M        269          Kings Bay   
3     Levy  2018-02-01 00:00:00  MNW18026    M        202  Black Point Swamp   
4  Broward  2018-02-01 00:00:00  MSE18024    U        275    Port Everglades   
5  Brevard  2018-02-02 00:00:00  MEC18013    F        280       Indian River   

               18                                   21  
1            City                       Probable Cause  
2   Crystal River                       Natural: Other  
3       Cedar Key                 Natural: Cold Stress  
4  Ft. Lauderdale  Human Related: Watercraft Collision  
5   Port St. John                 Human Related: Other

In [10]:
# convert headers

headers = clean_df.iloc[0]
new_df = pd.DataFrame(data=clean_df.values[1:], columns=headers)
new_df.head()

1   County                 Date  Field ID Sex Size (cm)           Waterway  \
0   Citrus  2018-02-01 00:00:00  MNW18021   M       269          Kings Bay   
1     Levy  2018-02-01 00:00:00  MNW18026   M       202  Black Point Swamp   
2  Broward  2018-02-01 00:00:00  MSE18024   U       275    Port Everglades   
3  Brevard  2018-02-02 00:00:00  MEC18013   F       280       Indian River   
4  Brevard  2018-02-02 00:00:00  MEC18014   M       209   Eau Gallie River   

1            City                       Probable Cause  
0   Crystal River                       Natural: Other  
1       Cedar Key                 Natural: Cold Stress  
2  Ft. Lauderdale  Human Related: Watercraft Collision  
3   Port St. John                 Human Related: Other  
4       Melbourne                 Natural: Cold Stress

In [28]:
# I'm going to remove that unnecessary timestamp eventually

In [29]:
new_df.tail()

1       County                 Date  Field ID Sex Size (cm)       Waterway  \
734  St. Lucie  2018-12-30 00:00:00  MSE18098   M       201   Taylor Creek   
735        Lee  2018-12-30 00:00:00  MSW18340   F       140  Matlacha Pass   
736     Monroe  2018-12-30 00:00:00  MSW18341   M       265    Florida Bay   
737    Collier  2018-12-30 00:00:00  MSW18343   M       208  Tamiami Canal   
738    Collier  2018-12-31 00:00:00  MSW18342   F       154    Roberts Bay   

1          City                       Probable Cause  
734  Ft. Pierce         Undetermined: Too Decomposed  
735  Cape Coral                Perinatal (<= 150 cm)  
736    Flamingo  Human Related: Watercraft Collision  
737     Ochopee                 Natural: Cold Stress  
738      Naples         Undetermined: Too Decomposed

In [30]:
new_df = new_df.to_csv('manatee_mortality_2018.csv', index=False)

#### Can drop rows 744 & 745 because they're just a summary of the deaths for all of 2018:
* 824 total deaths
* 123 Watercraft Deaths (Florida only)

### Mortality comparisons for the time period from 01/01 to the ending date (final + preliminary data)


In [31]:
comparison_df = df.iloc[747:, 0:]
comparison_df

0   1   2   3   4           5    6                7    8   \
747            Year NaN NaN NaN NaN  Watercraft  NaN  Flood Gate/Lock  NaN   
748            2018 NaN NaN NaN NaN   123 (15%)  NaN                6  NaN   
749            2017 NaN NaN NaN NaN   111 (21%)  NaN                3  NaN   
750            2016 NaN NaN NaN NaN   106 (20%)  NaN                2  NaN   
751            2015 NaN NaN NaN NaN    86 (21%)  NaN                5  NaN   
752            2014 NaN NaN NaN NaN    69 (19%)  NaN                3  NaN   
753            2013 NaN NaN NaN NaN     73 (9%)  NaN                5  NaN   
754  5-year average NaN NaN NaN NaN    89 (18%)  NaN                3  NaN   

      9   ...           16       17   18           19            20   21  \
747  NaN  ...  Cold Stress  Natural  NaN  Unrecovered  Undetermined  NaN   
748  NaN  ...           79      228  NaN          105           120  NaN   
749  NaN  ...           30       93  NaN           26           114  NaN   
750  NaN  ...           24       86  NaN           36           113  NaN   
751  NaN  ...           18       44  NaN           31            95  NaN   
752  NaN  ...           26       29  NaN           17            80  NaN   
753  NaN  ...           40      198  NaN          100           125  NaN   
754  NaN  ...           27       90  NaN           42           105  NaN   

                     22     23  24  25  
747  Undetermined Other  Total NaN NaN  
748                  29    824 NaN NaN  
749                  29    538 NaN NaN  
750                  34    520 NaN NaN  
751                  24    405 NaN NaN  
752                  38    371 NaN NaN  
753                 148    830 NaN NaN  
754                  54    532 NaN NaN  

[8 rows x 26 columns]

In [32]:
comparison_df.isnull().sum()

0     0
1     8
2     8
3     8
4     8
5     0
6     8
7     0
8     8
9     8
10    8
11    0
12    8
13    8
14    0
15    8
16    0
17    0
18    8
19    0
20    0
21    8
22    0
23    0
24    8
25    8
dtype: int64

In [34]:
df.iloc[747:, 8:]

8    9    10           11   12   13         14   15           16  \
747  NaN  NaN  NaN  Other Human  NaN  NaN  Perinatal  NaN  Cold Stress   
748  NaN  NaN  NaN           21  NaN  NaN        113  NaN           79   
749  NaN  NaN  NaN           15  NaN  NaN        117  NaN           30   
750  NaN  NaN  NaN            6  NaN  NaN        113  NaN           24   
751  NaN  NaN  NaN           11  NaN  NaN         91  NaN           18   
752  NaN  NaN  NaN            9  NaN  NaN        100  NaN           26   
753  NaN  NaN  NaN           12  NaN  NaN        129  NaN           40   
754  NaN  NaN  NaN           10  NaN  NaN        110  NaN           27   

          17   18           19            20   21                  22     23  \
747  Natural  NaN  Unrecovered  Undetermined  NaN  Undetermined Other  Total   
748      228  NaN          105           120  NaN                  29    824   
749       93  NaN           26           114  NaN                  29    538   
750       86  NaN           36           113  NaN                  34    520   
751       44  NaN           31            95  NaN                  24    405   
752       29  NaN           17            80  NaN                  38    371   
753      198  NaN          100           125  NaN                 148    830   
754       90  NaN           42           105  NaN                  54    532   

     24  25  
747 NaN NaN  
748 NaN NaN  
749 NaN NaN  
750 NaN NaN  
751 NaN NaN  
752 NaN NaN  
753 NaN NaN  
754 NaN NaN

In [35]:
# drop null columns
comparison_df.dropna(axis=1, inplace=True)
comparison_df

0           5                7            11         14  \
747            Year  Watercraft  Flood Gate/Lock  Other Human  Perinatal   
748            2018   123 (15%)                6           21        113   
749            2017   111 (21%)                3           15        117   
750            2016   106 (20%)                2            6        113   
751            2015    86 (21%)                5           11         91   
752            2014    69 (19%)                3            9        100   
753            2013     73 (9%)                5           12        129   
754  5-year average    89 (18%)                3           10        110   

              16       17           19            20                  22  \
747  Cold Stress  Natural  Unrecovered  Undetermined  Undetermined Other   
748           79      228          105           120                  29   
749           30       93           26           114                  29   
750           24       86           36           113                  34   
751           18       44           31            95                  24   
752           26       29           17            80                  38   
753           40      198          100           125                 148   
754           27       90           42           105                  54   

        23  
747  Total  
748    824  
749    538  
750    520  
751    405  
752    371  
753    830  
754    532

In [36]:
comparison_df.to_csv('five_year_mortality_comparison.csv', index=False)